### ISSUE TITLE
Add the ticket on behalf of anybody

### ISSUE THREAD

**Issue by wptech68** [nanodesigns/nanosupport#70]
Im trying to add a ticket on behalf of a specific end user (subscriber) who can add a ticket by himself, but it doesnt show in the users list, only those with support agents and adminis are showed in the list.
Is this the correct behaviour? Thanks

_Originally posted by @wptech68 in https://github.com/nanodesigns/nanosupport/issue_comments#issuecomment-446673925_

**Comment by wptech68** [comment]
Followed the guidelines, but I cannot see the full list of end users in the dropdown list.
In your example if "niyoti" is a standard user (subscriber in wordpress) I cannot replicate because the list just shows the support agents and admin. 
In my installation, all end users can add a ticket my themselves from the front end, then ticket is assigned by admin to a support agent, so both end user who created the ticket and support agent can see and reply the ticket as it should be, all work fine there.
However, I as the admin cannot add a new ticket on behalf of a user because they are not listed in the dropdown author list.
have no idea about how capabilities may conflict here since all users all allowed to use the nanosupport by themselves, thanks anyway

**Comment by mayeenulislam** [comment]
What version of NanoSupport you are using?<br>
Because the scenario you are saying was present before `v0.4.1` and should not be _that_ way in the latest version.

In our window all the users are now available with `v0.5.0`:<br>
![all-users-available](https://user-images.githubusercontent.com/4551598/49952190-aad8ae00-ff25-11e8-8b6a-e63df2a15578.png)

**Comment by wptech68** [comment]
0.5.0 for sure
now I see that I have 6 support agents and 1 admin, but the list only shows the admin and 3 agents (not 6)
so not all support seekers are listed apart of not listing any end user.
It should be some role/permission problem, but is strange since all users are allowed to work with the plugin posting new tickets, looks is something only related to permissions for the "add on behalf..." function

**Comment by mayeenulislam** [comment]
We're believing in you. But if it is possible, then we would love to have an access to the site under question, so that we can inspect whether there is an issue there?

If, and only if, it is possible, you can create an account for us with: **info [at] nanodesignsbd [dot] com**

**Comment by wptech68** [comment]
will do it if I cannot find the problem.
Now after looking more I can confirm that only the users with  "author" or "admin" privilegies at WP are the ones that are listed in the dropdown list, but not the "subscribers", independently if they are support seekers or not.
For example, a standar user with "subscriber" role is NOT listed in the dropdown list even if is also a support agent , but any user with author or admin status is listed

**Comment by mayeenulislam** [comment]
@wptech68 You are right. We just replicated the issue in our end. Yes, not all the users are visible in the dropdown list. And we also [found the reason](https://github.com/nanodesigns/nanosupport/blob/v0.5.1/includes/ns-cpt-nanosupport.php#L466-L485).

By default, only 4 user roles are available in the dropdown and `subscriber` is not there. But there's a filter hook so that you can add any other user role to the dropdown if you think you need them.

Filter hook: `nanosupport_assigned_user_role` is used to add `subscriber`. The following code can be placed in your theme's `functions.php`:

```php
<?php
add_filter('nanosupport_assigned_user_role', function($array) {
    return $array = array_merge($array, array(
    	'subscriber' => __('Subscriber')
    ));
});
```

We will update the documentation with the information as soon as possible, inshALLAH

**Comment by wptech68** [comment]
worked perfect, thanks
